pkg/runtime/processor/state/state.go

```diff
@@ -86,16 +86,17 @@ func (s *state) Init(ctx context.Context, comp compapi.Component) error {
 	secretStoreName := s.meta.AuthSecretStoreOrDefault(&comp)
 
 	secretStore, _ := s.compStore.GetSecretStore(secretStoreName)
-	encKeys, encErr := encryption.ComponentEncryptionKey(comp, secretStore)
-	if encErr != nil {
+	encKeys, err := encryption.ComponentEncryptionKey(comp, secretStore)
+	if err != nil {
 		diag.DefaultMonitoring.ComponentInitFailed(comp.Spec.Type, "creation", comp.ObjectMeta.Name)
 		return rterrors.NewInit(rterrors.CreateComponentFailure, fName, err)
 	}
 
 	if encKeys.Primary.Key != "" {
 		ok := encryption.AddEncryptedStateStore(comp.ObjectMeta.Name, encKeys)
 		if ok {
-			log.Infof("automatic encryption enabled for state store %s", comp.ObjectMeta.Name)
+			log.Infof("Automatic encryption enabled for state store %s", comp.ObjectMeta.Name)
+			log.Warnf("WARNING: ⚠️ Automatic state store encryption should never be used to encrypt more than 2^32 values. Encrypting more than that can cause the private key to be exposed.")
 		}
 	}
```

Comment:
How about:

```suggestion
	if encKeys.Primary.Key != "" {
		ok := encryption.AddEncryptedStateStore(comp.ObjectMeta.Name, encKeys)
		if ok {
			log.Infof("Automatic encryption enabled for state store %s", comp.ObjectMeta.Name)
			log.Warnf("WARNING: ⚠️ Automatic state store encryption should never be used to encrypt more than 2^32 values. Encrypting more than that can cause the private key to be exposed.")
		}
	} else {
		log.Infof("Automatic encryption disabled for state store %s", comp.ObjectMeta.Name)
	}
```

Reply:
I don't think it is _super_ necessary to have this as a log warning over putting it in the docs as it could seem a bit spooky for users even when doing the right thing.

If we do want to log a message it should be at the INFO level, and we should use a human number to put 2^32 in context (`over 4 billion values`). We should also state that the key can be rotated.

Reply:
> I don't think it is _super_ necessary to have this as a log warning over putting it in the docs as it could seem a bit spooky for users even when doing the right thing.
> 
> If we do want to log a message it should be at the INFO level, and we should use a human number to put 2^32 in context (`over 4 billion values`). We should also state that the key can be rotated.

Agree with proposed changes

Reply:
I also agree:
Info level logging, and using the human friendly "4 billion". But in the docs, let's actually make it a **warning** not just some note / info thing.

Reply:
>  it could seem a bit spooky for users even when doing the right thing.

Note that 2^32 is a number the NIST sets as a general guideline. The problem is that the nonce is random and 96-bit, and you must never use the same nonce twice with the same key. Given the birthday problem, the NIST stated that after 2^32 the _likelihood_ of nonce reuse (with random nonces) is too high. However, a collision could happen after 2^80 or even just 2^1.

> Info level logging, and using the human friendly "4 billion"

I think warning is important, but agree "4 billion" is better

Reply:
Please, change to Info, change the text to 4 billion and remove the warning sign emoji. Other than that LGTM and we can merge

Reply:
Done